src/components/LoginModal/LoginModal.js

```diff
@@ -63,12 +63,8 @@ const LoginModal = ({
             onChange={handleChange}
           />
           <span className="modal__error">{errors.password}</span>
+          <span className="modal__error-login">{serverError}</span>
         </label>
-        {serverError && (
-          <span className="modal__error-unavailable">
-            Incorrect email or password
-          </span>
-        )}
       </div>
     </ModalWithForm>
```

Comment:
**Could be improved**

If it’s interesting for you here is a way to make a special wrapper for any `modal` with cross button, `Escape` and `overlay` listeners inside. You should create component `Modal`, store it in `Modal.js`. 


```jsx

import { useEffect } from "react";

export const Modal = ({ name, onClose, children }) => {
  // here is `useEffect` for the `Escape` listener
  useEffect(() => {
    // we should define the handler inside `useEffect`, so that it wouldn’t lose the reference to be able to remove it
    const handleEscape = (e) => {
      if (e.key === "Escape") {
        onClose();
      }
    };

    document.addEventListener("keydown", handleEscape);
    // don’t forget to remove the listener in the `clean-up` function
    return () => document.removeEventListener("keydown", handleEscape);
  }, [onClose]);

  // here is the overlay handler
  const handleOverlay = (e) => {
    if (e.target === e.currentTarget) {
      onClose();
    }
  };

  // then we add the main wrapper with class `modal`
  return (
    <div className={`modal modal_type_${name}`} onClick={handleOverlay}>
      {/* the container for the contents */}
      <div className="modal__container">
        {/* here will be anything you add as `children`*/}
        {children}
        {/* add the close button */}
        <button className="modal__close" type="button" onClick={onClose} />
      </div>
    </div>
  );
};



```


And now you can use `Modal` for any popup in the project:  `ItemModal`  and `ModalWithForm` 

```jsx


function ModalWithForm({ name, onClose, ...props}) {
  return (
    <Modal name={name} onClose={onClose}>
        <h2 className='popup__title'>{props.title}</h2>


```

You can make also `Input` for inputs and `Form` for any form in the project.

Reply:
Thank you for showing this to me! I submitted the main corrections, but I will continue to work on improving the project by implementing this. I think this will be very useful for future projects as well. I didn't realize I could break it down even further like this, this is great!